napari/plugins/npe2api.py

```diff
@@ -90,12 +117,14 @@ def iter_napari_plugin_info() -> (
 
         # TODO: once the new version of npe2 is out, this can be refactored
         # to all the metadata includes the conda and pypi versions.
-        extra_info = {
-            "home_page": _info.get("home_page", ""),
-            "pypi_versions": _info.pop("pypi_versions"),
-            "conda_versions": _info.pop("conda_versions"),
-        }
+        extra_info = _ExtraDict(
```

Comment:
... As here. With a dictionary, any Python user knows what you are making in these lines. With `_ExtraDict`, they have to go find what the hell you're talking about.

Reply:
If the type name contains `Dict` then user should assume that it is dict with more constraints. 

I have removed `_ExtraDict` but for me it end user with less information.